cardano-node/cardano-node.cabal

```diff
@@ -156,7 +156,7 @@ library
                       , cardano-ledger-binary
                       , cardano-ledger-byron
                       -- TODO: remove constraint at next ledger bump
```

Comment:
Cannot this TODO be removed?

Reply:
Good catch. However, I think that comment is probably referring to the major bump to `1.18`. I'll take that out in the 10.2 PR (#6050). This one is just cherry-picking what was already done for the hotfix.